### PR TITLE
Enable type checks for test files

### PR DESCRIPTION
**Pull request by rtktpeng**
Renaming `setupTests.js` to `setupTests.ts` seemed to make tsc aware of the `jest-enzyme` matchers. 

I was originally looking into https://github.com/kulshekhar/ts-jest, which I believe offers something stricter (e.g. you can't run jest when there is a type error). I'm however more used to the softer check: errors in IDEs and `tsc` output only; jest unaffected, which is the outcome of this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "types": "dist/ui.d.ts",
   "license": "MIT",
   "scripts": {
+    "check:types": "tsc --noEmit",
```
Comment (rtktpeng) [inline package.json:9]: While fixing type errors throughout the project, I needed to run type checks repeatedly without generating output.

File: src/components/menu/__tests__/Menu.spec.tsx [hunk 2]
```diff
@@ -40,6 +40,7 @@ describe('Menu', () => {
     );
 
     expect(
+      // @ts-ignore
       wrapper.find('ContextProvider').prop('value').selectedItem
```
Comment (rtktpeng) [inline src/components/menu/__tests__/Menu.spec.tsx:44]: I may be looking into fixing all the suppressions in tests later.

File: package.json (hunk 1 above)
Comment (rtkaaho) [reply package.json:9]: would it be worth running this in the pre-commit hooks or something? Or maybe running it in the github workflow?

Comment (rtktpeng) [reply package.json:9]: I guess it's already part of the github workflow (any type error will break the automated checks) I just like a standalone command.

File: tsconfig.rollup.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "src/**/*.stories.tsx",
     "src/**/story.components.tsx",
     "src/**/*.spec.tsx",
-    "src/**/*.spec.ts"
+    "src/**/*.spec.ts",
+    "setupTest.ts"
```
Comment (rtktpeng) [inline tsconfig.rollup.json:25]: This address https://github.com/rtkwlf/ui/pull/115/checks?sha=a7c3d8a7f6ae9978e2ef204721507c89a1261027#step:8:14
I guess rollup is looking for `.ts`, so `setupTest.js` was ignored as desired previously. After the rename, we need to exclude it manually.

**Pull request merged**